### DISCUSSION TITLE
How to disable OAI(origin access identity) of cloudfront-origin in CDK v2?

### DISCUSSION THREAD

**Discussion by cckn** [aws/aws-cdk#19675] [Q&A]
I'm using KMS to encrypt objects in my S3 bucket and serving them from cloudfront.

I need to disable OAI in cloudfront-origin to use KMS.

I'm using CDK 2.4.0 and I couldn't find `how to disable OAI` in the CDK documentation.

If originAccessIdentity is not entered, the default for OAI in CDK v1 is `OAI is not used`.
[cdk v1 originaccessidentity](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cloudfront.S3OriginConfig.html#originaccessidentity
)

On the other hand, in CDK v2, if you do not input OAI, `CDK creates and uses OAI`.
[cdk v2 originaccessidentity](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront_origins.S3OriginProps.html#originaccessidentity
)

I can change that setting in the console, but I have to change it every time the CDK is finished deploying.
It is not easy to use only that stack as CDK v1 because all of the existing infrastructure is using CDK v2.

Is there any way to disable OAI in CDK v2?

```ts
 const origin = new cloudfrontOrigins.S3Origin(bucket)

 const cloudfrontDistributionBehaviorOptions: cdk.aws_cloudfront.BehaviorOptions = {
    origin,
    viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
    trustedKeyGroups: [keyGroup],
    edgeLambdas: [
        {
            functionVersion,
                eventType: cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST,
        },
    ],
 }

 this.cloudfrontDistribution = new cloudfront.Distribution(this, `cloudfront-distribution`, {
    defaultBehavior: {
        ...cloudfrontDistributionBehaviorOptions,
    },
    additionalBehaviors: {
        "*.m3u8": {
            ...cloudfrontDistributionBehaviorOptions,
            cachePolicy: cloudfront.CachePolicy.CACHING_DISABLED,
        },
    },
 })
```

**Comment by peterwoodworth** [comment]
You can remove the OAI config on your distribution with [escape hatches](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html). Here's an example:

```ts
        const dist = new Distribution(this, 'Dist', {
            defaultBehavior: {
                origin,
            }
        })

        const cfdist = dist.node.defaultChild as CfnDistribution;
        cfdist.addPropertyDeletionOverride('DistributionConfig.Origins.0.S3OriginConfig.OriginAccessIdentity')
```

But also, nothing about using cdk v2 should prevent you from using the `CloudFrontWebDistribution` construct if that's easier. You can create both [`CloudFrontWebDistribution`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cloudfront.CloudFrontWebDistribution.html) and [`Distribution`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cloudfront.Distribution.html) in both v1 and v2

Hope this helps

**Reply by danielbayerlein** [reply]
Does not work:
```js
cfdist.addPropertyDeletionOverride('DistributionConfig.Origins.0.S3OriginConfig')
```

Works for me:
```js
cfdist.addPropertyOverride('DistributionConfig.Origins.0.S3OriginConfig.OriginAccessIdentity', '');
```

**Reply by peterwoodworth** [reply]
Thanks, I had the wrong path. I've updated my comment and it should be good now.

**Comment by belinde** [comment]
I too had to use `addPropertyOverride` with an empty string, and it works OK. But the OriginAccessIdentity is still created, even if it's not attached to the distribution. Is there a way to remove it from the stack?
